src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -989,11 +1014,26 @@ internal string ConvertPSObjectToCSV(PSObject mshObject, IList<string> propertyN
                     _outputString.Append(_delimiter);
                 }
 
-                // If property is not present, assume value is null and skip it.
-                if (mshObject.Properties[propertyName] is PSPropertyInfo property)
+                string value = null;
+                if (mshObject.BaseObject is IDictionary dictionary)
                 {
-                    var value = GetToStringValueForProperty(property);
+                    if (dictionary.Contains(propertyName))
```

Comment:
Can we use TryGet()?

Reply:
I always forget that's a thing. Will do, thanks!

Reply:
Checked the .NET API docs, but it looks like the non-generic `IDictionary` doesn't have `TryGetValue()` unfortunately:

https://docs.microsoft.com/en-us/dotnet/api/system.collections.idictionary?view=netcore-3.0#methods

That only comes with the generic version of IDictionary, which we could use but may have other implications... 🤔

Reply:
I see.

Does ExportCsvHelper.BuildPropertyNames() ensure that check is always true?

Reply:
No. There's no guarantee that each input object will have the same dictionary entries; BuildPropertyNames is only used to check the first input object. 

After that, the CSV cmdlets ignore missing properties & assume them to be $null.

Reply:
I hope nobody will process large IDictionary. :-(

Reply:
It is a possible problem, but I would guess it is only about as useful as someone serializing a PSCustomObject with a few hundred properties, so I don't think it's a particularly likely scenario.

Is there a simpler / more performant alternative to determining the presence of a key?

Reply:
Every class implements the interface in it _own way_. As result we can not write effective code for all the classes.